core/worker.go

```diff
@@ -976,6 +978,29 @@ func (w *worker) commit(env *environment, interval func(), update bool, start ti
 	return nil
 }
 
+// pendingBodyKey comprises all root fields of the header
+type pendingBodyKey struct {
+	UncleHash common.Hash
+	TxHash    common.Hash
+	EtxHash   common.Hash
+}
+
+// PendingBlockBodyKey takes a header and hashes all the Roots together
+// and returns the key to be used for the pendingBlockBodyCache.
+func (w *worker) PendingBlockBodyKey(header *types.Header) (hash common.Hash) {
+	nodeCtx := common.NodeLocation.Context()
+	hasher := blake3.New(32, nil)
```

Comment:
Don't we use keccak everywhere else? Is this the first place (outside of PoW) that uses Blake3? I'd prefer we be consistent and use the same hash function for all commitments, if thats the case.

Reply:
No. This is logic replicated from ‘SealHash’ function. we use blake3 hash there

Reply:
Okay. you mean outside PoW. i can use  keccack if that is the case. i will check that and let you know

Reply:
Using Keccack hash now